File: 프로그래머스/3/42884. 단속카메라/단속카메라.java
import java.util.*;
public class Solution {
    public int solution(int[][] routes) {
        int cnt = 0;
        int position = -30001;

        Arrays.sort(routes, (a, b) -> Integer.compare(a[1], b[1]));

        for (int[] route : routes) {
            if (position < route[0]) {
                position = route[1];
                cnt++;
            }
        }

        return cnt;
    }
}

File: 프로그래머스/3/42884. 단속카메라/단속카메라.js
function solution(routes) {
    let sortedRoutes = routes.sort((a, b) => a[1] - b[1]);
    
    let result = 0;
    let camera = -30001;
    
    sortedRoutes.forEach(([s, e]) => {
        if(s > camera){
            camera = e;
            result++;
        }
    })
    return result;
}